### PR TITLE
feat: integrate `RLN` UI

### PR DESCRIPTION
**Pull request by 0xisk**
## Explanation

This PR adds UI RLN Proof screens

## More Information

- Related #91 

## Screenshots/Screencaps

Screenshot_20230802_130746
Screenshot_20230802_130723

## Manual Testing Steps

- Open demo app local.
- Checkout RLN Proof generation options.
- Review the RLN Approve Proof screen.

## Pre-Merge Checklist

- [x] PR template is filled out
- [ ] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [x] PR is linked to the appropriate GitHub issue
- [x] PR has been added to the appropriate release Milestone

> PR template source from [github.com/MetaMask](https://github.com/MetaMask)

### CODE REVIEW

File: packages/types/tsconfig.json [hunk 1]
```diff
@@ -8,5 +8,10 @@
     "lib": ["es2020"]
   },
   "include": ["src"],
-  "exclude": ["node_modules", "dist"]
+  "exclude": ["node_modules", "dist"],
+  "references": [
+    {
+      "path": "../proofs/rln-proof"
+    }
+  ]
```
Comment (0xmad) [inline packages/types/tsconfig.json:16]: We don't need this in types

File: packages/types/src/proof/zkProof.ts [hunk 2]
```diff
@@ -13,14 +13,19 @@ export interface ZkInputs {
   merkleProofArtifactsOrStorageAddress: string | MerkleProofArtifacts;
 }
 
+// TODO: that is a redandant
 export interface ZKProofPayload {
```
Comment (0xmad) [inline packages/types/src/proof/zkProof.ts:17]: Why is it redundant?

**Review by 0xmad — APPROVED** [review]
@0xisk thanks, just some minor comments.

File: packages/types/src/proof/zkProof.ts (hunk 2 above)
Comment (0xisk) [reply packages/types/src/proof/zkProof.ts:17]: The intention is to optimize it later and not to have repeated properties inside types in general. As you can see that: 
```ts
  // types related to ZkInputs
  circuitFilePath: string;
  verificationKey: string;
  zkeyFilePath: string;
```

```ts
  // types related to Semaphore
  externalNullifier: string;
  signal: string;
```

```ts
  // types related to ZkMetadata
    urlOrigin: string;
```


```ts
  // types related to ZkInputs
  merkleStorageAddress?: string;
```

So there is a optimization needs to be happen here, and will check that in this PR: #673

**Pull request merged**